float64 max_dist
float64 min_dist